src/main/java/core/basesyntax/service/implementation/DecreaseOperationImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service.implementation;
+
+import core.basesyntax.dto.FruitRecordDto;
+import core.basesyntax.model.Storage;
+import core.basesyntax.service.FruitOperationHandler;
+
+public class DecreaseOperationImpl implements FruitOperationHandler {
+    @Override
+    public int apply(FruitRecordDto fruitRecordDto) {
+        Integer currentQuantity = Storage.getFruits().get(fruitRecordDto.getFruitName());
+        if (currentQuantity >= fruitRecordDto.getQuantity()) {
```

Comment:
It's better to use Optional for currentQuantity since you may get NPE here if map returns null